### DISCUSSION TITLE
MonthView.getter:willSelectDate can never be called

### DISCUSSION THREAD

**Discussion by nefch** [kvyatkovskys/KVKCalendar#276] [Q&A]
Have a look at `CalendarView.init(frame:date:style:years:)`. The code on line 91/92 will never assign the willSelectDate variable.

```
monthView.willSelectDate = { [weak self] (date) in
    self?.delegate?.willSelectDate(date, type: .month)
}
```

That's because `self.delegate` is not yet set. The delegate can only be set **after** the constructor has run.

This means that `CalendarDelegate.willSelectDate(_:type:)` will never be called.

**Comment by kvyatkovskys** [comment]
hmm, I'll check this part

**Reply by kvyatkovskys** [reply]
`monthView.willSelectDat` works if you set `style.month.isPagingEnabled = true` in the current version of library

**Reply by nefch** [reply]
I do not see any difference. `CalendarDelegate.willSelectDate(_:type:)` is still not getting called when I scroll the month view.

**Reply by kvyatkovskys** [reply]
hmm interesting.... works fine for me 🤔

**Reply by nefch** [reply]
I'm using your example code to test this.
I add `style.month.isPagingEnabled = true` to `KVKCalendarSettings.createCalendarStyle()`.
I add `func willSelectDate(_ date: Date, type: CalendarType) { print("\(date), \(type)") }` to ViewController.swift.
`willSelectDate` is never called.

Maybe you could integrate `willSelectDate` into your example code?

**Reply by kvyatkovskys** [reply]
hmm interesting... I'll check this case again
